pkg/volume/hostpath/host_path.go

```diff
@@ -489,11 +489,13 @@ func makeDir(pathname string) error {
 // If pathname already exists, whether a file or directory, no error is returned.
 func makeFile(pathname string) error {
 	f, err := os.OpenFile(pathname, os.O_CREATE, os.FileMode(0644))
-	defer f.Close()
 	if err != nil {
 		if !os.IsExist(err) {
 			return err
 		}
 	}
+	if f != nil {
```

Comment:
if f is nil, then OpenFile should return err != nil?

But I am not sure why it checks !os.IsExist(err) earlier?

Reply:
@jingxu97 
From existing logic, f.Close() should be called if we don't return on line 493.
Meaning, the err may be nil, maybe not nil.

For IsExist() error, we cannot say the makeFile fails. So nil is returned to the caller.